crates/accelerate/src/sabre_swap/mod.rs

```diff
@@ -57,10 +60,80 @@ pub enum Heuristic {
     Decay,
 }
 
-struct TrialResult {
-    out_map: HashMap<usize, Vec<[usize; 2]>>,
-    gate_order: Vec<usize>,
-    layout: NLayout,
+/// A container for Sabre mapping results.
+#[pyclass(module = "qiskit._accelerate.sabre_swap")]
+#[derive(Clone, Debug)]
+pub struct SabreResult {
+    #[pyo3(get)]
+    pub map: SwapMap,
+    pub node_order: Vec<usize>,
+    #[pyo3(get)]
+    pub node_block_results: NodeBlockResults,
+}
+
+#[pymethods]
+impl SabreResult {
+    #[getter]
+    fn node_order(&self, py: Python) -> PyObject {
+        self.node_order.to_pyarray(py).into()
+    }
+}
+
+#[pyclass(module = "qiskit._accelerate.sabre_swap")]
```

Comment:
Do you think we should explicitly call this a mapping? Just so nothing assume it's a sequence (the pyo3 handling around this is a bit weird because the python c api is a bit ambiguous about the difference as there are overlapping slots). See https://pyo3.rs/v0.19.1/class/protocols#mapping--sequence-types for more details

```suggestion
#[pyclass(mapping, module = "qiskit._accelerate.sabre_swap")]
```

Reply:
Done in 3f7425b.